x/poolmanager/router.go

```diff
@@ -577,7 +577,10 @@ func (k Keeper) ListPoolsByDenom(
 
 		var poolsByDenom []types.PoolI
 		for _, pool := range currentModulePools {
-			poolDenoms := pool.GetPoolDenoms(ctx)
+			poolDenoms, err := poolModule.GetPoolDenoms(ctx, pool.GetId())
```

Comment:
can we add test case where we have error from this?

Reply:
I have been thinking about this logic and I wonder if instead of returning an error should any pools that dont expose the denoms be skipped? considering the case for future cosm wasm pools, we would be relying that every single potential contract has the query implemented (correctly)

Reply:
Yeah, I think they should be skipped, filing as a new issue!